js/mirador_viewer.js

```diff
@@ -12,10 +12,13 @@
      */
     Drupal.behaviors.Mirador = {
         attach: function (context, settings) {
+            settings.mirador.instances = {}
             Object.entries(settings.mirador.viewers).forEach(entry => {
-              const [base, values] = entry;
-              once('mirador-viewer', base, context).forEach(() =>
-                Mirador.viewer(values, window.miradorPlugins || {})
+              const [viewerId, settings] = entry;
```

Comment:
It looks like this would be masking the `settings` variable as passed into the `attach()`), so later setting (as in `settings.mirador.instances`) probably wouldn't be in the correct place? Like, it would be setting a reference to the viewer in a spot where a reference to the viewer is necessary to get it in the first place, relative to the viewer? As in, below, trying to set to `drupalSettings.mirador.viewers.{viewerId}.mirador.instances.{viewerId}`?

I guess, if we _do_ want it under `drupalSettings`, we could look at just setting it under `viewers.{viewerId}.instance` or something, to track the instance for each viewer; though, would probably want to look at moving the configuration of the instance into a new subkey, so yeah, maybe the parallel object structure by ID would make sense?

Reply:
Good catch. I just got this deployed in staging and caught this. But your new code changes fix this regardless